### ISSUE TITLE
Cannot launch: Issue with import lmfit.models

### ISSUE THREAD

**Issue by aconceicao** [nexpy/nexpy#379]
I'm trying to run the package nexpy by I always get the following error:
```from lmfit.models import lmfit_models
ImportError: cannot import name 'lmfit_models' from 'lmfit.models' (C:\Users\concea\AppData\Local\Continuum\anaconda3\lib\site-packages\lmfit\models.py)'''


I have installed the lmfit package but it did not help. Have anyone from you faced the same problem?
Thanks.

**Comment by rayosborn** [comment]
Can you let me know what version of lmfit you are running? `lmfit_models` was introduced in v1.0.2, which was released in February, 2021. If you just installed NeXpy, I'm a little surprised by the error, because the lmfit dependency specified versions greater than 1.0.3.

**Comment by aconceicao** [comment]
Hi Raymond. My ``lmfit`` version was 1.0.0. Besides it the installation was smoothly done. I have updated the version of the ``lmfit`` and reinstalled ``nexpy``. Now it is working. However, it crashes if I open a large file (5 dimensions) of ~8 Gb.

**Comment by rayosborn** [comment]
There seem to be two issues here. One is how you were able to install NeXpy with an outdated lmfit version, since both pip and conda specify >v1.0.3 as a requirement. Can you describe how you installed NeXpy, so that I can prevent this in the future? 

The second is why it crashed opening the file. Can you post the file somewhere so that I can download it? The size is not particularly large by NeXpy standards, and, in any case, the nexusformat API does not load large arrays. I have not seen NeXpy crash when opening a file for a very long time, so there may be another issue with the file itself.

**Comment by aconceicao** [comment]
I just installed using conda ``conda install -c conda-forge nexpy`` on a windows machine and no warning was raised up. By updating the ``lmfit`` and reinstalling ``nexpy`` it launched. In this same topic, I tried to install it on a Debian 10 machine. Again, the installation was smoothly performed. But by launching the GUI the problem was with the ``qtconsole``. In this case, updating the ``qtconsole`` package and reinstal ``nexpy`` did not work.
Concerning the crash when opening a large dataset, I do not believe it is a problem with the file since I can open it with other packages like ``silx view``. I can upload the file to a cloud and share with you. Can you please, inform me your e-mail address?

**Comment by rayosborn** [comment]
I just checked the conga-forge recipe and the lmfit dependency is correctly specified, so it is concerning that it was ignored when you installed NeXpy. I will have to look into whether this is a known issue.

I should have asked what you meant when you said NeXpy crashed. Was there literally a core dump or did it just generate an exception? If it is the latter, NeXpy stores the stack trace in a log file, accessible from the Window menu. Please post the trace here if it's available.

I wasn't automatically assuming that your file was corrupt, but if there is something in the way it was written that causes the nexusformat API problems, I will certainly fix it. You can use rosborn@anl.gov as my email address.

**Comment by prjemian** [comment]
One way an application using h5py could fail is if the data file uses a
compression codec that is not installed. Might that be the case?

**Comment by rayosborn** [comment]
That's an excellent suggestion. I should check to see how `nexusformat` and/or `h5py` handles that. If I can find an installer for a non-standard codec, I will try that out on one of our Linux machines and then copy the file to my Mac. It's worth investigating even if it isn't the reason for this particular problem.

**Comment by prjemian** [comment]
The h5plugin package provided many. All you need to do is import it into
the module. It will install entry points to handle various additional
compression codecs.

**Comment by rayosborn** [comment]
Although adding support for `hdf5plugin` is something I am actively pursuing, I don't think it can have been responsible for NeXpy crashing if that is indeed what happened. Trying to read a compressed field with an unsupported filter just generates an exception when trying to access the field values.

**Comment by aconceicao** [comment]
Hi @rayosborn . I've sent you an e-mail with the link to download the file I mentioned. What I mean by the crash is that the program shut down once I try to go through the file tree.  For the lower-dimension files, Nexpy works fine.

**Comment by rayosborn** [comment]
@aconceicao, as I wrote by email, I was unable to download the file because of the network dropping the connection. Until we sort this out, I have released v1.0.0rc2 of the `nexusformat` package, which enables support for `hdf5plugin`. Please install it before trying NeXpy again and let me know if it makes a difference.
```
$ pip install nexusformat==1.0.0rc2
```
It may be irrelevant to this issue, but it's a useful feature.

**Comment by rayosborn** [comment]
@aconceicao, I have managed to download your data and have been able to view it without problem. 
![merged_data](https://user-images.githubusercontent.com/1193959/203631168-1bf41f3d-8c00-4930-af2d-1db88b4666a6.png)
As you know, following @prjemian's suggestion, I added the [hdf5plugin](https://github.com/silx-kit/hdf5plugin) as an import in `nexusformat v1.0.0rc2`, just in case your data required a non-standard compression filter. It turns out that your multi-dimensional data does indeed use a compression filter, so presumably the import worked. However, I don't really understand how it worked because, when I try to find out which filter was used, I get the following:
```
>>> merged_data.nxfile['processed/result/data'].id.get_create_plist().get_filter(0)
ValueError: Filter number is invalid (filter number is invalid)
```
So apparently, `nexusformat` (or rather `h5py` + `hdf5plugin`) decompressed the data, even though the filter number was invalid. 

I had to use such an odd command to extract the compression, because there is a limitation in `h5py`, which return None, when the compression is non-standard. 
```
>>> merged_data.nxfile['processed/result/data'].compression is None
True
```
This limitation is being discussed in a [current h5py issue](https://github.com/h5py/h5py/issues/2161).

Please let me know if installing `nexusformat v1.0.0rc2` fixes your problem.

**Comment by aconceicao** [comment]
@rayosborn thanks. After updating the `nexusformat` package I could open the file. However, it slowed down the PC. This does not happen when I open smaller files. Do you have any idea?

**Comment by rayosborn** [comment]
I don't understand why opening a file should cause a slowdown. The memory footprint of a loaded file, even if it contains large datasets, is very small, unless you have set the `maxsize` and `memory` parameters to very large values. Please post the output from `nxgetconfig()`. 

Is this slowdown consistently reproducible? Operating systems can suddenly slow down for a variety of reasons, so could it have been a coincidence?

**Comment by rayosborn** [comment]
Talking about coincidences, it appears that no compression was applied to `merged_data['processed/result/data']` after all (see [here](https://github.com/h5py/h5py/issues/2161) for details), so there is no reason it should have been fixed by installing v1.0.0rc2, which only added imports to hdf5plugin. I can only assume that the crash and now your slowdown are pointing to the same issue with your PC. What memory does your PC have? Perhaps it is necessary to reduce the nexusformat memory limits from the 2GB default. To reduce it to 1GB, type `nxgetconfig(memory=1000)` before loading the file or set it in the NeXpy preferences and save as default.

**Comment by rayosborn** [comment]
@aconceicao, I suggest we close this issue for now, since I believe you are saying that the original problem, i.e., crashing when opening a file, no longer occurs. If the slowdown is still a reproducible problem, please open another issue, preferably giving details of the nexusformat and NeXpy versions you are running, and the output to `nxgetconfig`.

From your shell:
```
$ nexpy -v
nexpy v1.0.0rc3
$ nexusformat -v
nexusformat v1.0.0rc4
```
Within the Python REPL:
```
>>> from nexusformat.nexus import nxgetconfig
>>> nxgetconfig()
{'compression': 'gzip', 'encoding': 'utf-8', 'lock': 0, 'lockexpiry': 28800, 'lockdirectory': None, 'maxsize': 10000, 'memory': 2000, 'recursive': False}
```
